### PR TITLE
fixed Mono -> Flux -> Mono back and forth conversion

### PR DESCRIPTION
**Pull request by moderakh**
in different layers we were unnecessarily translating Mono to Flux and back to Mono.

When we know the return type is single we should use Mono and not translate back and forth from Mono to Flux and back to Mono again.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/rx/examples/multimaster/Helpers.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ static public String createDatabaseUri(String databaseName) {
                                 }
                             }
 
-                            return Flux.error(e);
+                            return Mono.error(e);
                         }
                 ).map(ResourceResponse::getResource).single();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/rx/examples/multimaster/Helpers.java]: We don't need this `.single()` anymore, right ?
Here and every other method in this class.

File: sdk/cosmos/azure-cosmos-examples/src/test/java/com/azure/cosmos/rx/examples/CollectionCRUDAsyncAPITest.java [hunk 2]
```diff
@@ -242,7 +243,7 @@ public void createCollection_toBlocking_CollectionAlreadyExists_Fails() {
      */
     @Test(groups = "samples", timeOut = TIMEOUT)
     public void transformObservableToCompletableFuture() throws Exception {
-        Flux<ResourceResponse<DocumentCollection>> createCollectionObservable = client
+        Mono<ResourceResponse<DocumentCollection>> createCollectionObservable = client
                 .createCollection(getDatabaseLink(), collectionDefinition, null);
         CompletableFuture<ResourceResponse<DocumentCollection>> future = createCollectionObservable.single().toFuture();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-examples/src/test/java/com/azure/cosmos/rx/examples/CollectionCRUDAsyncAPITest.java]: Same here, unnecessary conversion to `.single()`

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java [hunk 3]
```diff
@@ -94,7 +94,7 @@ public RxClientCollectionCache(ISessionContainer sessionContainer,
             retryPolicyInstance.onBeforeSendRequest(request);
         }
 
-        Flux<RxDocumentServiceResponse> responseObs = this.storeModel.processMessage(request);
+        Mono<RxDocumentServiceResponse> responseObs = this.storeModel.processMessage(request);
         return responseObs.map(response -> BridgeInternal.toResourceResponse(response, DocumentCollection.class)
                 .getResource()).single();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java]: Please remove unnecessary `.single()`

File: sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/rx/examples/multimaster/Helpers.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-examples/src/main/java/com/azure/cosmos/rx/examples/multimaster/Helpers.java]: as discussed offline. they are intentionally kept as they will do correctness validation to ensure Mono is not empty.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxClientCollectionCache.java]: as discussed offline. they are intentionally kept as they will do correctness validation to ensure Mono is not empty.

**Comment by moderakh** [comment]
> Thanks for bringing this change in.
> Please remove few more `.single()` instances.

as discussed offline. they are intentionally kept as they will do correctness validation to ensure Mono is not empty.

**Pull request merged**